test/ci_support/cdash_analyze_and_report_random_failures_UnitTests.py

```diff
@@ -0,0 +1,288 @@
+
+    def test_no_slash(self):
+        test_dict = { 'buildSummaryLink':'buildid' }
+        self.assertEqual(
+            getBuildIdFromTest(test_dict), 'buildid')
+
+
+#############################################################################
+#
+# System-level tests for cdash_analyze_and_report_random_failures.py
+#
+#############################################################################
+
+
+class test_cdash_analyze_and_report_random_failures(unittest.TestCase):
+    def cdash_analyze_and_report_random_failures_run_case(
+        self,
+        expectedRtnCode,
+        stdoutRegexList,
+        htmlFileRegexList,
+        extraCmndLineOptionsList=None,
+        verbose=False,
+        debugPrint=False,
+    ):
+        if not extraCmndLineOptionsList:
+            extraCmndLineOptionsList = []
+
+        htmlFileName = "htmlFile.html"
+        htmlFileAbsPath = os.getcwd()+"/"+htmlFileName
+
+        cmnd = (
+            ciSupportDir
+            + "/cdash_analyze_and_report_random_failures.py"
+            + " --cdash-project-name='Project Name'"
+            + " --group-name='Group Name'"
+            + " --initial-nonpassing-test-filters='initial_nonpassing_test_filters'"
+            + " --cdash-site-url='https://something.com/cdash'"
+            + " --reference-date=2018-10-28"
+            + " --write-email-to-file="+htmlFileName
+            + " "
+            + " ".join(extraCmndLineOptionsList)
+        )
+
+        stdoutFile = "stdout.out"
+        stdoutFileAbsPath = os.getcwd() + "/" + stdoutFile
+        rtnCode = CDQAR.echoRunSysCmnd(
+            cmnd, throwExcept=False, outFile=stdoutFile, verbose=verbose
+        )
+
+        stdout = ""
+        try:
+            with open(stdoutFile, "r") as inf:
+                stdout = inf.read()
+        except Exception:
+            print("WARNING: No stdout available from this test")
+
+        # UNCOMMENT THIS TO SEE OUTPUT FROM THE SCRIPT (Can use the test to help develop)
+        # print(stdout)
+
+        self.assertEqual(rtnCode, expectedRtnCode, "Failed with stdout: " + stdout)
+
+        # Grep stdout for expected list of strings
+        assertListOfRegexsFoundInListOfStrs(self, stdoutRegexList,
+            stdout.splitlines(), stdoutFileAbsPath, debugPrint=debugPrint)
+
+        # Grep written HTML file for expected strings
+        try:
+            with open(htmlFileName, 'r') as htmlFile:
+                htmlFileStrList = htmlFile.read().split("\n")
+        except Exception:
+            print("WARNING: HTML file not available for this test: "+htmlFileAbsPath)
+
+        assertListOfRegexsFoundInListOfStrs(self, htmlFileRegexList,
+            htmlFileStrList, htmlFileAbsPath, debugPrint=debugPrint)
+
+
+    def setUp(self):
+        self.test_dir = TemporaryDirectory()
+        os.chdir(self.test_dir.name)
+
+    def tearDown(self):
+        self.test_dir.remove()
+
+
+    # Test the random failure case of a single initial test containing
```

Comment:
Move this to a Python docstring ('''stuff''' right after the function declaration)